### PR TITLE
[branch-2.0](binlog) Support drop view binlog (#39781)

### PR DESCRIPTION
**Pull request by wyxxxcat**
pick #39781

### CODE REVIEW

**Comment by wyxxxcat** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 1]
```diff
@@ -2054,29 +2054,24 @@ private Status allTabletCommitted(boolean isReplay) {
...
+                }
+                if (tableType == TableType.OLAP && backupTableInfo != null) {
+                    // drop the non restored partitions.
+                    dropNonRestoredPartitions(db, (OlapTable) table, backupTableInfo);
+                } else if (isCleanTables) {
+                    // otherwise drop the entire table.
+                    LOG.info("drop non restored table {}({}). {}", tableName, tableId, this);
+                    boolean isView = false;
+                    boolean isForceDrop = false; // move this table into recyclebin.
+                    env.getInternalCatalog().dropTableWithoutCheck(db, table, isView, isForceDrop);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:2074]: @wyxxxcat This patch is staled, please update.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 2]
```diff
@@ -916,23 +916,26 @@ public void dropTable(DropTableStmt stmt) throws DdlException {
                 }
             }
 
-            dropTableInternal(db, table, stmt.isForceDrop());
+            dropTableInternal(db, table, stmt.isView(), stmt.isForceDrop(), watch, costTimes);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:919]: only pick the necessary parameters.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 3]
```diff
@@ -916,23 +916,26 @@ public void dropTable(DropTableStmt stmt) throws DdlException {
...
         } catch (UserException e) {
             throw new DdlException(e.getMessage(), e.getMysqlErrorCode());
         } finally {
             db.writeUnlock();
         }
-        LOG.info("finished dropping table: {} from db: {}, is force: {}", tableName, dbName, stmt.isForceDrop());
+        watch.stop();
+        costTimes.put("6:total", watch.getTime());
+        LOG.info("finished dropping table: {} from db: {}, is view: {}, is force: {}, cost: {}",
+                tableName, dbName, stmt.isView(), stmt.isForceDrop(), costTimes);
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:928]: should merge this conflict manually, eg: 

```
LOG.info("finished dropping table: {} from db: {}, is force: {}, is view: {}", tableName, dbName, stmt.isForceDrop(), stmt.isView());
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 4]
```diff
@@ -942,7 +945,8 @@ public void dropTableWithoutCheck(Database db, Table table, boolean forceDrop) t
     }
 
     // Drop a table, the db lock must hold.
-    private void dropTableInternal(Database db, Table table, boolean forceDrop) throws DdlException {
+    private void dropTableInternal(Database db, Table table, boolean isView, boolean forceDrop,
+            StopWatch watch, Map<String, Long> costTimes) throws DdlException {
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:949]: remove `StopWatch watch, Map<String, Long> costTimes`

**Review by w41ter — CHANGES_REQUESTED** [review]

**Comment by w41ter** [comment]
close in favor #44112

**Pull request closed without merging**